modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java

```diff
@@ -22,7 +22,7 @@
 public class CodegenParameter implements IJsonSchemaValidationProperties {
     public boolean isFormParam, isQueryParam, isPathParam, isHeaderParam,
             isCookieParam, isBodyParam, hasMore, isContainer,
-            secondaryParam, isCollectionFormatMulti, isPrimitiveType, isModel, isExplode;
+            secondaryParam, isCollectionFormatMulti, isPrimitiveType, isModel, isExplode, jsonEncoding;
```

Comment:
In OAI 3.0, encoding can be specified besides `application/json`, so this method seems to have low extensibility.
This change will affect the entire openapi-generator and will require a maintainers review.

cc. @wing328 @jimschubert

Reply:
I agree. If you need custom logic, I think a vendor extension would be more appropriate.

Reply:
Not quite sure I follow - from what I read about vendor extensions on [the docs] (https://swagger.io/docs/specification/openapi-extensions/), they are meant for _functionality that is not covered by the standard OpenAPI Specification_ .  This isn't custom logic, this is functionality that is part of the [OAI specification](https://swagger.io/docs/specification/describing-request-body/), and is currently ignored by the OAI generator.

I do agree that other encoding types can be specified and should be supported as well, but since the objects already include a ToJson(), it seemed like an easy win for those that need it.

Reply:
@milesressler please follow `isJson` approach in Dart client generator: https://github.com/OpenAPITools/openapi-generator/blob/8d6286dfae0c37d53ce9d27cd00b8e8ef87bcfe1/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java#L272

Let me know if you need help. You can PM me via Slack (mentioned in the readme)